2023-12-20T12:34:56.145355+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [8] [ERROR] Exception in worker process
2023-12-20T12:34:56.145368+00:00 app[web.1]: Traceback (most recent call last):
2023-12-20T12:34:56.145370+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
2023-12-20T12:34:56.145382+00:00 app[web.1]:     worker.init_process()
2023-12-20T12:34:56.145383+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/workers/base.py", line 134, in init_process
2023-12-20T12:34:56.145385+00:00 app[web.1]:     self.load_wsgi()
2023-12-20T12:34:56.145386+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
2023-12-20T12:34:56.145386+00:00 app[web.1]:     self.wsgi = self.app.wsgi()
2023-12-20T12:34:56.145386+00:00 app[web.1]:                 ^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145387+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/base.py", line 67, in wsgi
2023-12-20T12:34:56.145387+00:00 app[web.1]:     self.callable = self.load()
2023-12-20T12:34:56.145387+00:00 app[web.1]:                     ^^^^^^^^^^^
2023-12-20T12:34:56.145387+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
2023-12-20T12:34:56.145388+00:00 app[web.1]:     return self.load_wsgiapp()
2023-12-20T12:34:56.145388+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145388+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
2023-12-20T12:34:56.145389+00:00 app[web.1]:     return util.import_app(self.app_uri)
2023-12-20T12:34:56.145389+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145389+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/util.py", line 371, in import_app
2023-12-20T12:34:56.145390+00:00 app[web.1]:     mod = importlib.import_module(module)
2023-12-20T12:34:56.145390+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145391+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/importlib/__init__.py", line 90, in import_module
2023-12-20T12:34:56.145393+00:00 app[web.1]:     return _bootstrap._gcd_import(name[level:], package, level)
2023-12-20T12:34:56.145393+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145393+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
2023-12-20T12:34:56.145393+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
2023-12-20T12:34:56.145394+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
2023-12-20T12:34:56.145394+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
2023-12-20T12:34:56.145394+00:00 app[web.1]:   File "<frozen importlib._bootstrap_external>", line 994, in exec_module
2023-12-20T12:34:56.145395+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
2023-12-20T12:34:56.145395+00:00 app[web.1]:   File "/app/app.py", line 12, in <module>
2023-12-20T12:34:56.145395+00:00 app[web.1]:     model = joblib.load('model_scoring_LGBM.joblib') # lightgbm model
2023-12-20T12:34:56.145395+00:00 app[web.1]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145396+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/joblib/numpy_pickle.py", line 658, in load
2023-12-20T12:34:56.145396+00:00 app[web.1]:     obj = _unpickle(fobj, filename, mmap_mode)
2023-12-20T12:34:56.145396+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145397+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
2023-12-20T12:34:56.145397+00:00 app[web.1]:     obj = unpickler.load()
2023-12-20T12:34:56.145397+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145397+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1205, in load
2023-12-20T12:34:56.145397+00:00 app[web.1]:     dispatch[key[0]](self)
2023-12-20T12:34:56.145397+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1530, in load_stack_global
2023-12-20T12:34:56.145398+00:00 app[web.1]:     self.append(self.find_class(module, name))
2023-12-20T12:34:56.145398+00:00 app[web.1]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145398+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1572, in find_class
2023-12-20T12:34:56.145398+00:00 app[web.1]:     __import__(module, level=0)
2023-12-20T12:34:56.145398+00:00 app[web.1]: ModuleNotFoundError: No module named 'lightgbm'
2023-12-20T12:34:56.145424+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [7] [ERROR] Exception in worker process
2023-12-20T12:34:56.145424+00:00 app[web.1]: Traceback (most recent call last):
2023-12-20T12:34:56.145424+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
2023-12-20T12:34:56.145424+00:00 app[web.1]:     worker.init_process()
2023-12-20T12:34:56.145424+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/workers/base.py", line 134, in init_process
2023-12-20T12:34:56.145424+00:00 app[web.1]:     self.load_wsgi()
2023-12-20T12:34:56.145424+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
2023-12-20T12:34:56.145425+00:00 app[web.1]:     self.wsgi = self.app.wsgi()
2023-12-20T12:34:56.145425+00:00 app[web.1]:                 ^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145425+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/base.py", line 67, in wsgi
2023-12-20T12:34:56.145425+00:00 app[web.1]:     self.callable = self.load()
2023-12-20T12:34:56.145425+00:00 app[web.1]:                     ^^^^^^^^^^^
2023-12-20T12:34:56.145425+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
2023-12-20T12:34:56.145425+00:00 app[web.1]:     return self.load_wsgiapp()
2023-12-20T12:34:56.145425+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145426+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
2023-12-20T12:34:56.145426+00:00 app[web.1]:     return util.import_app(self.app_uri)
2023-12-20T12:34:56.145426+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145426+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/util.py", line 371, in import_app
2023-12-20T12:34:56.145426+00:00 app[web.1]:     mod = importlib.import_module(module)
2023-12-20T12:34:56.145426+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145426+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/importlib/__init__.py", line 90, in import_module
2023-12-20T12:34:56.145427+00:00 app[web.1]:     return _bootstrap._gcd_import(name[level:], package, level)
2023-12-20T12:34:56.145427+00:00 app[web.1]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145427+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
2023-12-20T12:34:56.145427+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
2023-12-20T12:34:56.145427+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
2023-12-20T12:34:56.145427+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
2023-12-20T12:34:56.145427+00:00 app[web.1]:   File "<frozen importlib._bootstrap_external>", line 994, in exec_module
2023-12-20T12:34:56.145428+00:00 app[web.1]:   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
2023-12-20T12:34:56.145428+00:00 app[web.1]:   File "/app/app.py", line 12, in <module>
2023-12-20T12:34:56.145428+00:00 app[web.1]:     model = joblib.load('model_scoring_LGBM.joblib') # lightgbm model
2023-12-20T12:34:56.145428+00:00 app[web.1]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145428+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/joblib/numpy_pickle.py", line 658, in load
2023-12-20T12:34:56.145428+00:00 app[web.1]:     obj = _unpickle(fobj, filename, mmap_mode)
2023-12-20T12:34:56.145428+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145429+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
2023-12-20T12:34:56.145429+00:00 app[web.1]:     obj = unpickler.load()
2023-12-20T12:34:56.145429+00:00 app[web.1]:           ^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145429+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1205, in load
2023-12-20T12:34:56.145429+00:00 app[web.1]:     dispatch[key[0]](self)
2023-12-20T12:34:56.145429+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1530, in load_stack_global
2023-12-20T12:34:56.145430+00:00 app[web.1]:     self.append(self.find_class(module, name))
2023-12-20T12:34:56.145430+00:00 app[web.1]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.145430+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/pickle.py", line 1572, in find_class
2023-12-20T12:34:56.145430+00:00 app[web.1]:     __import__(module, level=0)
2023-12-20T12:34:56.145430+00:00 app[web.1]: ModuleNotFoundError: No module named 'lightgbm'
2023-12-20T12:34:56.145518+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [8] [INFO] Worker exiting (pid: 8)
2023-12-20T12:34:56.145593+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [7] [INFO] Worker exiting (pid: 7)
2023-12-20T12:34:56.304471+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [2] [ERROR] Worker (pid:8) exited with code 3
2023-12-20T12:34:56.306795+00:00 app[web.1]: [2023-12-20 12:34:56 +0000] [2] [ERROR] Worker (pid:7) exited with code 3
2023-12-20T12:34:56.306836+00:00 app[web.1]: Traceback (most recent call last):
2023-12-20T12:34:56.306840+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 209, in run
2023-12-20T12:34:56.306988+00:00 app[web.1]:     self.sleep()
2023-12-20T12:34:56.307036+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 360, in sleep
2023-12-20T12:34:56.307136+00:00 app[web.1]:     ready = select.select([self.PIPE[0]], [], [], 1.0)
2023-12-20T12:34:56.307213+00:00 app[web.1]:             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-12-20T12:34:56.307222+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
2023-12-20T12:34:56.307273+00:00 app[web.1]:     self.reap_workers()
2023-12-20T12:34:56.307286+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 530, in reap_workers
2023-12-20T12:34:56.307363+00:00 app[web.1]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
2023-12-20T12:34:56.307392+00:00 app[web.1]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2023-12-20T12:34:56.307395+00:00 app[web.1]: 
2023-12-20T12:34:56.307396+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2023-12-20T12:34:56.307396+00:00 app[web.1]: 
2023-12-20T12:34:56.307396+00:00 app[web.1]: Traceback (most recent call last):
2023-12-20T12:34:56.307405+00:00 app[web.1]:   File "/app/.heroku/python/bin/gunicorn", line 8, in <module>
2023-12-20T12:34:56.307430+00:00 app[web.1]:     sys.exit(run())
2023-12-20T12:34:56.307453+00:00 app[web.1]:              ^^^^^
2023-12-20T12:34:56.307461+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/wsgiapp.py", line 67, in run
2023-12-20T12:34:56.307491+00:00 app[web.1]:     WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
2023-12-20T12:34:56.307509+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/base.py", line 236, in run
2023-12-20T12:34:56.307556+00:00 app[web.1]:     super().run()
2023-12-20T12:34:56.307571+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/app/base.py", line 72, in run
2023-12-20T12:34:56.307598+00:00 app[web.1]:     Arbiter(self).run()
2023-12-20T12:34:56.307617+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 229, in run
2023-12-20T12:34:56.307659+00:00 app[web.1]:     self.halt(reason=inst.reason, exit_status=inst.exit_status)
2023-12-20T12:34:56.307679+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 342, in halt
2023-12-20T12:34:56.307739+00:00 app[web.1]:     self.stop()
2023-12-20T12:34:56.307753+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 396, in stop
2023-12-20T12:34:56.307813+00:00 app[web.1]:     time.sleep(0.1)
2023-12-20T12:34:56.307826+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
2023-12-20T12:34:56.307868+00:00 app[web.1]:     self.reap_workers()
2023-12-20T12:34:56.307879+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.12/site-packages/gunicorn/arbiter.py", line 530, in reap_workers
2023-12-20T12:34:56.307950+00:00 app[web.1]:     raise HaltServer(reason, self.WORKER_BOOT_ERROR)
2023-12-20T12:34:56.307965+00:00 app[web.1]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2023-12-20T12:34:56.437155+00:00 heroku[web.1]: Process exited with status 1
2023-12-20T12:34:56.457419+00:00 heroku[web.1]: State changed from up to crashed